diff --git a/external/bsd/flex/dist/src/flexdef.h b/external/bsd/flex/dist/src/flexdef.h
index 72f7ab1..aec38c2 100644
--- a/external/bsd/flex/dist/src/flexdef.h
+++ b/external/bsd/flex/dist/src/flexdef.h
@@ -40,6 +40,13 @@
 #include <config.h>
 #endif
 
+#ifndef __BEGIN_DECLS
+#define __BEGIN_DECLS
+#endif
+#ifndef __END_DECLS
+#define __END_DECLS
+#endif
+
 /* AIX requires this to be the first thing in the file.  */
 #if !defined(__GNUC__) && !defined(__lint__)
 # if HAVE_ALLOCA_H
diff --git a/sys/lib/libunwind/DwarfParser.hpp b/sys/lib/libunwind/DwarfParser.hpp
index 6e8ef2c..8a5ef81 100644
--- a/sys/lib/libunwind/DwarfParser.hpp
+++ b/sys/lib/libunwind/DwarfParser.hpp
@@ -383,7 +383,7 @@ CFI_Parser<A, R>::parseInstructions(A &addressSpace, pint_t instructions,
       break;
     case DW_CFA_remember_state:
       // avoid operator new, because that would be an upward dependency
-      entry = (PrologInfoStackEntry *)malloc(sizeof(PrologInfoStackEntry));
+      entry = (PrologInfoStackEntry *)std::malloc(sizeof(PrologInfoStackEntry));
       if (entry == NULL)
         return false;
 
@@ -398,7 +398,7 @@ CFI_Parser<A, R>::parseInstructions(A &addressSpace, pint_t instructions,
         PrologInfoStackEntry *top = rememberStack;
         *results = top->info;
         rememberStack = top->next;
-        free((char *)top);
+        std::free((char *)top);
       }
       break;
     case DW_CFA_def_cfa:
